Sources/MapboxMaps/Style/Style.swift

```diff
@@ -721,6 +725,71 @@ public final class Style: StyleProtocol {
         }
     }
 
+    /// Adds an image to be used in the style.
+    ///
+    /// This API can also be used for
+    /// updating an image. If the image id was already added, it gets replaced
+    /// by the new image.
+    ///
+    /// The image can be used in
+    /// [`icon-image`](https://www.mapbox.com/mapbox-gl-js/style-spec/#layout-symbol-icon-image),
+    /// [`fill-pattern`](https://www.mapbox.com/mapbox-gl-js/style-spec/#paint-fill-fill-pattern), and
+    /// [`line-pattern`](https://www.mapbox.com/mapbox-gl-js/style-spec/#paint-line-line-pattern).
+    ///
+    /// - Parameters:
+    ///   - image: Image to add.
+    ///   - id: ID of the image.
+    ///   - sdf: Option to treat whether image is SDF(signed distance field) or not.
+    ///         Setting this to `true` allows template images to be recolored. The
+    ///         default value is `false`.
+    ///   - content: An array of four numbers, with the first two specifying the
+    ///         left, top corner, and the last two specifying the right, bottom
+    ///         corner. If present, and if the icon uses icon-text-fit, the
+    ///         symbol's text will be fit inside the content box.
+    ///
+    /// - Throws:
+    ///     An error describing why the operation was unsuccessful.
+    public func addImage(_ image: UIImage, id: String, sdf: Bool = false, content: ImageContent? = nil) throws {
+        try addImage(image, id: id, stretchX: [], stretchY: [])
+    }
+
+    /// Adds a resizable image to be used in the style.
+    /// X- and Y-stretches will be calculated from `UIImage.capInsets`
+    ///
+    /// - Parameters:
+    ///   - image: Image to add.
+    ///   - id: ID of the image.
+    ///   - sdf: Option to treat whether image is SDF(signed distance field) or not.
+    ///         Setting this to `true` allows template images to be recolored. The
+    ///         default value is `false`.
+    ///   - content: An array of four numbers, with the first two specifying the
+    ///         left, top corner, and the last two specifying the right, bottom
+    ///         corner. If present, and if the icon uses icon-text-fit, the
+    ///         symbol's text will be fit inside the content box.
+    ///
+    /// - Throws:
+    ///     An error describing why the operation was unsuccessful.
+    public func addImageResizable(_ image: UIImage, id: String, sdf: Bool = false, content: ImageContent? = nil) throws {
+        guard image.capInsets != .zero else {
+            Log.warning(forMessage: "The supplied image is missing cap insets and might be resized incorrectly.")
+            try addImage(image, id: id, stretchX: [], stretchY: [])
+            return
+        }
```

Comment:
This guard statement makes me think that we don't need a separate methods. 
It looks like we can introduce united `addImage` method and highlight in the description that this method supports resizable images.

Reply:
Agreed. This is what I had in mind: https://github.com/mapbox/mapbox-maps-ios/pull/1269#pullrequestreview-941701627

Every UIImage has a non-optional capInsets property. It just defaults to zero for non-resizable images.